src/mame/sega/sega_ferie.cpp

```diff
@@ -0,0 +1,688 @@
+	}
+
+	required_device<t6a84_device> m_maincpu;
+	optional_device<palette_device> m_palette;
+	required_device<screen_device> m_screen;
+	required_device<ram_device> m_ram;
+
+	required_ioport m_io_buttons;
+	required_ioport m_io_pen_x;
+	required_ioport m_io_pen_y;
+	required_ioport m_io_pen_y_rescale;
+
+	uint8_t *m_rom;
+
+	uint8_t m_irq_ctrl;
+	uint8_t m_irq_status;
+	uint8_t m_irq_vector;
+
+	uint8_t m_rtc;
+
+	std::unique_ptr<uint8_t[]> m_lcd_dots;
+	uint8_t m_lcd_x;
+	uint8_t m_lcd_y;
+	uint8_t m_lcd_i;
+	uint8_t m_lcd_mode;
+
+	uint8_t m_tablet_cmd;
+	uint8_t m_tablet_cmd_prev;
+	uint8_t m_tablet_data_i;
+	uint16_t m_tablet_reply_state;
+	bool m_is_tablet_cmd_bit_set;
+
+	uint8_t m_button_state;
+	uint8_t m_pen_state;
+	uint8_t m_pen_target;
+};
+
+void sega_ferie_state::machine_start()
+{
+	uint8_t *rom = memregion("mask_rom")->base();
+	m_rom = reinterpret_cast<uint8_t *>(rom);
+	uint8_t *ram = m_ram->pointer();
+
+	m_maincpu->space(AS_PROGRAM).install_rom(0x00000, 0x7ffff, rom);
+	m_maincpu->space(AS_PROGRAM).install_ram(0x80000, 0x87fff, ram);
+	m_maincpu->space(AS_DATA).install_ram(0x00000, 0x7ffff, rom);
+	m_maincpu->space(AS_DATA).install_ram(0x80000, 0x87fff, ram);
+	m_maincpu->space(t6a84_device::AS_STACK).install_ram(0x00000, 0x7ffff, rom);
+	m_maincpu->space(t6a84_device::AS_STACK).install_ram(0x80000, 0x87fff, ram);
+
+	m_maincpu->space(AS_PROGRAM).install_read_handler(0x0000, 0xffff,
+			read8sm_delegate(*this, FUNC(sega_ferie_state::code_r)));
+	m_maincpu->space(AS_DATA).install_readwrite_handler(0x0000, 0xffff,
+			read8sm_delegate(*this, FUNC(sega_ferie_state::data_r)),
+			write8sm_delegate(*this, FUNC(sega_ferie_state::data_w)));
+	m_maincpu->space(t6a84_device::AS_STACK).install_readwrite_handler(0x0000, 0xffff,
+			read8sm_delegate(*this, FUNC(sega_ferie_state::stack_r)),
+			write8sm_delegate(*this, FUNC(sega_ferie_state::stack_w)));
+
+	m_maincpu->space(AS_IO).install_readwrite_handler(0xe4, 0xe4,
+			read8smo_delegate(*this, NAME([] () { return 0; })),
+			write8smo_delegate(*this, FUNC(sega_ferie_state::lcd_ctrl_w)));
+	m_maincpu->space(AS_IO).install_write_handler(0xe5, 0xe5,
+			write8smo_delegate(*this, FUNC(sega_ferie_state::lcd_data_w)));
+
+	m_maincpu->space(AS_IO).install_readwrite_handler(0xeb, 0xeb,
+			read8smo_delegate(*this, NAME([] () { return 0; /* TODO: On/Off button state */ })),
+			write8smo_delegate(*this, NAME([this] (uint8_t data) { m_irq_ctrl = data; })));
+
+	m_maincpu->space(AS_IO).install_read_handler(0xf4, 0xf4,
+			read8smo_delegate(*this, NAME([this] () { return m_rtc; })));
+
+	m_maincpu->space(AS_IO).install_read_handler(0xf5, 0xf5,
+			read8smo_delegate(*this, NAME([this] () { return m_io_buttons->read() & 0x3f; })));
+	m_maincpu->space(AS_IO).install_write_handler(0xf6, 0xf6,
+			write8smo_delegate(*this, FUNC(sega_ferie_state::tablet_ctrl_w)));
+	m_maincpu->space(AS_IO).install_read_handler(0xf7, 0xf7,
+			read8smo_delegate(*this, FUNC(sega_ferie_state::tablet_data_r)));
```

Comment:
Why do you need to install all this stuff here rather than using configured address maps?

Reply:
Not needed, I assumed it would override what I had defined on the processor side, but they actually get merged. I've refactored the IO space, should I try the same for the others?